src/B2_build_movie_db_framework.sql

```diff
@@ -0,0 +1,41 @@
+CREATE TABLE "user" (
+    user_id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    username VARCHAR(50) UNIQUE NOT NULL,
+    password VARCHAR(255) NOT NULL,
+    email VARCHAR(100) UNIQUE NOT NULL
+);
+
+CREATE TABLE search_history (
+    search_id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    user_id INTEGER REFERENCES "user"(user_id) ON DELETE CASCADE,
+    status VARCHAR(50),
+    note TEXT
+);
+
+CREATE TABLE bookmarks (
+    bookmark_id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    user_id INTEGER REFERENCES "user"(user_id) ON DELETE CASCADE,
+    media_id INTEGER REFERENCES media(id) ON DELETE CASCADE,
+    note TEXT
+);
+
+CREATE TABLE completed (
+    completed_id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    bookmark_id INTEGER REFERENCES bookmarks(bookmark_id) ON DELETE CASCADE,
+    completed_date DATE NOT NULL,
+    rewatchability INTEGER CHECK (rewatchability >= 1 AND rewatchability <= 10)
```

Comment:
Would suggest that we just use a scale from 1-5 on this as it doesn't make sense to have such a detailed rewatchability scale. I imagine the numbers meaning its something like `Very Low`, `Low`, `Medium`, `High` and `Very High`.

Reply:
Yes, this makes sense.